src/renderer/services/chain/fees/approve.ts

```diff
@@ -0,0 +1,27 @@
+import * as FP from 'fp-ts/lib/function'
+import * as Rx from 'rxjs'
+import * as RxOp from 'rxjs/operators'
+
+import { observableState } from '../../../helpers/stateHelper'
+import * as ETH from '../../ethereum'
+import { ApproveParams } from '../../ethereum/types'
+import { ApproveFeeHandler } from '../types'
+
+// state for reloading approve fees
+const { get$: reloadApproveFee$, set: reloadApproveFee } = observableState<ApproveParams | undefined>(undefined)
+
+const approveFee$: ApproveFeeHandler = (params) => {
+  return reloadApproveFee$.pipe(
+    RxOp.debounceTime(300),
+    RxOp.switchMap((approveParams) => {
+      return FP.pipe(
+        Rx.from(
+          // asset
+          ETH.approveTxFee$(approveParams || params)
+        )
+      )
+    })
+  )
+}
+
+export { reloadApproveFee, approveFee$ }
```

Comment:
No need to have `approveFee$` in `chain/fees/approve.ts`, since it's needed for ETH only. It can be removed, but `{ get$: reloadApproveFee$, set: reloadApproveFee }` needs to be move into `src/renderer/services/ethereum/fees.ts` before